jwst/photom/photom.py

```diff
@@ -691,7 +708,7 @@ def photom_io(self, tabdata, order=None):
             conversion = tabdata['photmj']              # unit is MJy
             if isinstance(self.input, datamodels.MultiSlitModel):
                 slit = self.input.slits[self.slitnum]
-                if self.exptype == 'NRS_MSASPEC':
+                if self.exptype in ['NRS_MSASPEC', 'NRS_FIXEDSLIT']:
```

Comment:
Is there an exptype where this conditional is intentionally excluding the following line?

Reply:
It's complicated logic that was imposed way back to handle the MOS mode master background subtraction processing in calspec2, where you occasionally have to force the step to use a given source_type (supplied as an argument to the step), rather than having it choose based on the existing `slit.source_type`. `self.source_type` is the value passed in as an argument and acts as an override. We need to use the `slit.source_type` for FS mode and this is the only if-block where that gets used, so I had to add it along with MOS. Other multiple-slit modes, like WFSS, don't need that logic (allowing for the possibility of an override). It's related to all the complicated logic to set `self.source_type` starting back up at https://github.com/spacetelescope/jwst/pull/7451/files#diff-b9378ce0b10971befd7443bb2b038a52954fb85c6a06906b7d0d6db5cb6a4057R137  At least that's my story and I'm sticking with it.